PowerShell/ScubaGear/Modules/Orchestrator.psm1

```diff
@@ -136,7 +139,25 @@ function Invoke-SCuBA {
         [Parameter(Mandatory = $false, ParameterSetName = 'Report')]
         [ValidateNotNullOrEmpty()]
         [string]
-        $OutReportName = "BaselineReports"
+        $OutReportName = "BaselineReports",
+
+        [Parameter(Mandatory = $true, ParameterSetName = 'Configuration')]
+        [ValidateScript({
+            if (-Not ($_ | Test-Path)){
+                throw "SCuBA configuration file or folder does not exist."
+            }
+            if (-Not ($_ | Test-Path -PathType Leaf)){
+                throw "SCuBA configuration Path argument must be a file."
+            }
+            return $true
+        })]
+        [System.IO.FileInfo]
+        $ConfigFilePath,
+
+        [Parameter(Mandatory  = $false, ParameterSetName = 'Configuration')]
+        [ValidateSet("json", "yaml", IgnoreCase = $false)]
+        [string]
+        $ConfigFormat = "json"
```

Comment:
So technically any valid JSON is valid YAML: https://alisoftware.github.io/yaml/2021/08/17/yaml-part1-json/

I'm not sure if ConvertFrom-Yaml will parse json-style structures correctly, but if it is a true yaml parser it should.  That would allow this to be simplified as the end user could just provide json and we parse it correctly.

Reply:
I took the conservative approach.  Used ConvertFrom-Json for JSON and ConvertFrom-Yaml for YAML.  We could explore only using ConvertFrom-Yaml for both formats - I'd want to do more testing of the ConvertFrom-Yaml before commiting to that.  It is only on version 0.4.2.  It does appear to be actively maintains (last updated 16 days ago).  Over 15M downloads of the current release version.

Reply:
Updated to use only ConvertFrom-Yaml and dropped conffiguration format parameter.  Updated unit test to verify each type (string, list, object, dictionary) work with either JOSN or YAML.